idea/idea-fir/src/org/jetbrains/kotlin/idea/core/overrideImplement/KtGenerateMembersHandler.kt

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.core.overrideImplement
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.SmartPsiElementPointer
+import com.intellij.psi.codeStyle.CodeStyleManager
+import org.jetbrains.kotlin.idea.core.insertMembersAfter
+import org.jetbrains.kotlin.idea.core.moveCaretIntoGeneratedElement
+import org.jetbrains.kotlin.idea.frontend.api.KtAnalysisSession
+import org.jetbrains.kotlin.idea.frontend.api.analyse
+import org.jetbrains.kotlin.idea.frontend.api.components.KtDeclarationRendererOptions
+import org.jetbrains.kotlin.idea.frontend.api.components.KtTypeRendererOptions
+import org.jetbrains.kotlin.idea.frontend.api.components.RendererModifier
+import org.jetbrains.kotlin.idea.frontend.api.symbols.KtCallableSymbol
+import org.jetbrains.kotlin.idea.frontend.api.tokens.HackToForceAllowRunningAnalyzeOnEDT
+import org.jetbrains.kotlin.idea.frontend.api.tokens.hackyAllowRunningOnEdt
+import org.jetbrains.kotlin.idea.util.application.runWriteAction
+import org.jetbrains.kotlin.psi.KtCallableDeclaration
+import org.jetbrains.kotlin.psi.KtClassBody
+import org.jetbrains.kotlin.psi.KtClassOrObject
+import org.jetbrains.kotlin.psi.KtDeclaration
+import org.jetbrains.kotlin.psi.psiUtil.endOffset
+import org.jetbrains.kotlin.psi.psiUtil.prevSiblingOfSameType
+import org.jetbrains.kotlin.psi.psiUtil.startOffset
+import org.jetbrains.kotlin.utils.addToStdlib.firstIsInstanceOrNull
+import org.jetbrains.kotlin.utils.addToStdlib.firstNotNullResult
+
+internal abstract class KtGenerateMembersHandler : AbstractGenerateMembersHandler<KtClassMember>() {
```

Comment:
The idea of Kt prefix is that the implementation is actually independent of FIR, it do not use FIR directly and works via HL API

Reply:
Yep Ilya is right to the point!

Reply:
Got it, thanks for clarifying. Is there a reason we shouldn't put these classes (except for the fix factories) in `idea-frontend-independent` then?

Also there is mention of "FIR" in this file but only in a comment.

Reply:
Thanks! I reworded the comment to make it clear that FiR is just an implementation detail. As to moving it out of idea-fir. I think that sounds good. I can't put it under independent because idea-frontend-api depends on it. How about putting it in idea-frontend-api? @darthorimar

Reply:
Discussed offline with Mark and it seems that idea-frontend-api actually doesn't need idea-frontend-independent. So should remove this line https://github.com/JetBrains/kotlin/blob/20c095a4f492efab9b6fd916711caf8814326909/idea/idea-frontend-api/build.gradle.kts#L12?